src/tiled/tileseteditor.cpp

```diff
@@ -435,8 +445,32 @@ void TilesetEditor::addTiles()
         QPixmap image;
     };
     QVector<LoadedFile> loadedFiles;
-
+    // If the tile is already in the tileset, warn user and confirm addition
+    bool dontAskAgain = false;
+    bool rememberOption = true;
     for (const QString &file : files) {
+        if (!(dontAskAgain && rememberOption) && hasTileInTileset(file, *tileset)) {
+            if (dontAskAgain)
+                continue;
+            QCheckBox *checkBox = new QCheckBox(tr("Apply this action to all tiles"));
+            QMessageBox warning(QMessageBox::Warning,
+                        tr("Add Tiles"),
+                        tr("Tile \"%1\" already exists in the tileset!").arg(file),
+                        QMessageBox::Yes | QMessageBox::No,
+                        mMainWindow->window());
+            warning.setDefaultButton(QMessageBox::Yes);
+            warning.setInformativeText(tr("Add anyway?"));
+            warning.setCheckBox(checkBox);
+            int warningBoxChoice = warning.exec();
+            if (checkBox->checkState() == Qt::Checked)
+                dontAskAgain = true;
+            if (warningBoxChoice != QMessageBox::Yes) {
+                rememberOption = false;
+                continue;
+            }
+            else
+                rememberOption = true;
```

Comment:
I think it's short and efficient, without compromising on readability. That's the kind of code I like. :-)

I don't like the name "rememberOption", but nothing better came to mind so far. For general readability, I considered whether using an `enum { Ask, Yes, No } addDuplicate;` (basically a tribool) could help. Feel free to explore that option if you want.